### PR TITLE
first dirty pass on refactoring

### PR DESCRIPTION
**Pull request by imoisejevs**
Not for merging.

Missing tests, docstrings, typehints
+ not implemented at node, cluster, job, replicaset levels

### CODE REVIEW

File: k8_kat/res/dep/kat_dep.py [hunk 1]
```diff
@@ -96,6 +89,17 @@ def aggregate_usage(self, fn: Callable) -> Optional[float]:
       return round(sum([fn(pod) for pod in self.pods()]), 3)
     except TypeError:
       return None
+
+  # todo interesting challenge - how do we add the running_normally decorator here?
```
Comment (xavier-r-millot) [inline k8_kat/res/dep/kat_dep.py:93]: excellent question. do you know what `metrics.k8s.io`'s  behavior is for a non-running pod?

File: k8_kat/res/dep/kat_dep.py [hunk 2]
```diff
@@ -96,6 +89,17 @@ def aggregate_usage(self, fn: Callable) -> Optional[float]:
...
+  def load_metrics(self):
+    return broker.custom.list_namespaced_custom_object(
+      group='metrics.k8s.io',
+      version='v1beta1',
+      namespace=self.namespace,
+      label_selector=label_conditions_to_expr(self.pod_select_labels.items(), []),
```
Comment (xavier-r-millot) [inline k8_kat/res/dep/kat_dep.py:99]: actually to address the above we could somewhat solve the problem by having a field_selector like `status.phase=Running`. Not totally thought through though

File: k8_kat/res/ns/kat_ns.py [hunk 3]
```diff
@@ -29,20 +29,29 @@ def is_work_ready(self) -> bool:
...
-  def pods(self):
-    from k8_kat.res.pod.kat_pod import KatPod
-    return KatPod.list(ns=self.name)
+  def load_metrics(self):
+    return broker.custom.list_namespaced_custom_object(
+      group='metrics.k8s.io',
+      version='v1beta1',
+      namespace=self.name,
+      plural='pods'
+    )['items']
```
Comment (xavier-r-millot) [inline k8_kat/res/ns/kat_ns.py:54]: safer to use `.get('items', [])` as k8s' cruelty knows now limits

Comment (xavier-r-millot) [reply k8_kat/res/ns/kat_ns.py:54]: haha get it? limits :)

File: k8_kat/utils/main/units.py [hunk 4]
```diff
@@ -3,10 +3,13 @@
 
 def find_multiplier_mapping(expr: str) -> Optional[Tuple[str, Union[int,float]]]:
   """Finds the correct multiplier for unit conversion."""
-  for candidate, multiplier in unit_map.items():
-    if expr.endswith(candidate):
-      return candidate, multiplier
-  return None
+  try:
```
Comment (xavier-r-millot) [inline k8_kat/utils/main/units.py:6]: what prompted this?

File: k8_kat/res/dep/kat_dep.py (hunk 1 above)
Comment (imoisejevs) [reply k8_kat/res/dep/kat_dep.py:93]: note: couple constructors in test_kat_pod for broken pods

**Pull request closed without merging**